Mathlib/ModelTheory/Equivalence.lean

```diff
@@ -257,6 +273,61 @@ theorem inf_iff_not_sup_not : (φ ⊓ ψ) ⇔[T] (φ.not ⊔ ψ.not).not :=
 
 end Formula
 
+namespace Theory
+
+variable (T) (α)
+
+/-- The type of formulas modulo a theory. -/
+protected def Formula := Quot (T.Iff : L.Formula α → _)
+
+variable {T} {α}
+
+namespace Formula
+
+instance : Coe (L.Formula α) (T.Formula α) := ⟨Quot.mk _⟩
+
+@[simps]
+instance : BooleanAlgebra (T.Formula α) where
```

Comment:
This is a special case of antisymmetrisation (see the similarly named type). Could you try reusing it to see whether it's practical to use? And if it isn't, whether there's anything we can do about it

Reply:
I had assumed that antisymmetrization would only be implemented for a `Preorder` instance. Now I see that it isn't, and this looks very promising - I'll give it a shot.

Reply:
One issue you might run into is the mismatch between `Iff` and `Imp and Imp`, but now that I'm thinking about it it seems that they actually match in your case?

Reply:
It looks not too bad to redefine `T.Iff` as `AntisymmRel T.Imp` - I think this might be better, so that equality in the quotient is definitionally `T.Iff`, but there will be a decent amount of `diff` in this PR if I do that.

Reply:
Is `T.Iff = AntisymmRel T.Imp` not defeq already?

Reply:
No - the proof is a few applications of `forall_and`.

Reply:
I've now given it a shot.